core/src/main/java/org/commonjava/maven/galley/internal/xfer/DownloadHandler.java

```diff
@@ -109,15 +109,15 @@ private Transfer joinOrStart( final ConcreteResource resource, final Transfer ta
                                   final EventMetadata eventMetadata )
             throws TransferException
     {
-        // if the target file already exists, skip joining.
-        if ( target.exists() )
-        {
-            return target;
-        }
-
         Future<DownloadJob> future;
         synchronized ( pending )
         {
+            // if the target file already exists, skip joining.
+            if ( target.exists() )
```

Comment:
It looks great. I think this solves the race condition you mentioned in jira comment. Btw, it would be easy to understand if we name the "pending" in the sync to something like DOWNLOAD_MUTEX, and make the pending itself a normal map. I will submit a PR for it.